File: bms/go/triangolatte/degreesToMeters/degreesToMeters.go
package test

import (
	"math"
)

var ()

const (
	originShift = 2.0 * math.Pi * 6378137 / 2.0
	math_Pi     = 3.14159265358979323846264338327950288419716939937510582097494459
)

func degreesToMeters(point Point) Point {
	return Point{
		point.X * originShift / 180.0,
		math.Log(math.Tan((90.0+point.Y)*math_Pi/360.0)) / (math_Pi / 180.0) * originShift / 180.0,
	}
}

type Point struct {
	X, Y float64
}

File: translations_only/fail/go/triangolatte/degreesToMeters/gemini/Restart/degreesToMeters.rs

use std::f64::consts::PI;

const ORIGIN_SHIFT: f64 = 2.0 * PI * 6378137.0 / 2.0;
const MATH_PI: f64 = 3.14159265358979323846264338327950288419716939937510582097494459;

pub fn degrees_to_meters(point: Point) -> Point {
    Point {
        x: point.x * ORIGIN_SHIFT / 180.0,
        y: (MATH_PI / 180.0) * ORIGIN_SHIFT / 180.0 * (point.y + 90.0).tan().ln(),
    }
}

#[derive(Debug, Clone, Copy)]
pub struct Point {
    pub x: f64,
    pub y: f64,
}
